### PR TITLE
Fix Installer Exception Handling v2 - TPROD-194

### LINKED ISSUE #9473: Installation throwing 500 error if credentials are incorrect
## Bug Description
I noticed while installing 3.2.1 today from the build packages that when my database credentials were incorrect during installation, instead of showing a warning it was crashing with a 500 error, and the following was in the error log:

```
[2020-12-05 12:08:47] mautic.WARNING: PHP Warning - Cannot use a scalar value as an array - in file /var/www/m/app/bundles/InstallBundle/Install/InstallService.php - at line 331 {"step":"[object] (Mautic\\InstallBundle\\Configurator\\Step\\DoctrineStep: {\"driver\":\"pdo_mysql\",\"host\":\"localhost\",\"table_prefix\":\"mau_\",\"port\":\"3306\",\"name\":\"redacted\",\"user\":\"redacted\",\"password\":\"redacted\",\"backup_tables\":0,\"backup_prefix\":\"bak_\"})","dbParams":{"driver":"pdo_mysql","host":"localhost","table_prefix":"mau_","port":"3306","name":"redacted","user":"redacted","password":"redacted","backup_tables":0,"backup_prefix":"bak_"},"translator":"[object] (Mautic\\CoreBundle\\Translation\\Translator: {})","messages":true,"schemaHelper":"[object] (Mautic\\InstallBundle\\Helper\\SchemaHelper: {})"} []
[2020-12-05 12:08:47] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\ConnectionException: "An exception occurred in driver: SQLSTATE[HY000] [1044] Access denied for user 'redacted'@'localhost' to database 'redacted'" at /var/www/m/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 93 {"exception":"[object] (Doctrine\\DBAL\\Exception\\ConnectionException(code: 0): An exception occurred in driver: SQLSTATE[HY000] [1044] Access denied for user 'redacted'@'localhost' to database 'redacted' at /var/www/m/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:93, Doctrine\\DBAL\\Driver\\PDOException(code: 1044): SQLSTATE[HY000] [1044] Access denied for user 'redacted'@'localhost' to database 'redacted' at /var/www/m/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:31, PDOException(code: 1044): SQLSTATE[HY000] [1044] Access denied for user 'redacted'@'localhost' to database 'redacted' at /var/www/m/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:27)"} []
```

This is a pretty ugly experience for the end-user, as they do not know what went wrong, and instead are greeted by our good friend the MautiBot.

Seems to relate to this line for the warning: https://github.com/mautic/mautic/blob/features/app/bundles/InstallBundle/Install/InstallService.php#L331

| Q   | A
| --- | ---
| Mautic version | 3.2.1
| PHP version | 7.4.3
| Browser | Chrome

### Steps to reproduce
1. Install Mautic and enter incorrect credentials when setting up the database connection
2. Note that you get a hard error, rather than a warning that the credentials may be invalid.
 
### Log errors

As above ^_-

### PR DESCRIPTION
**Pull request by dennisameling**
Successor of #9934

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features"
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | 
| Related developer documentation PR URL |
| Issue(s) addressed                     | Fixes #9473 

#### Description:
In 4.0alpha, going through the first step of the installer to configure and install the schema fails if using a db table prefix. It works fine without one but one throws an exception like `An error occurred while attempting to add default data: Table db_leads does not exist!`

To reproduce:
1. Go through the UI installer (delete or rename app/config/local.php)
2. Enter the database credentials with a db table prefix and proceed to the next step
3. Note the error
4. Repeat without a prefix and you'll be able to complete the installation
**Update**
@alanhartless wrote some patches that fix the issues on the UI.
 The bug exists on the CLI part of the installer, To reproduce the issue
1. Make sure the cache folder is empty.
2. Make sure `local.php` does not exist.
3. Make sure the database is empty.
3. Run the command to install Mautic `php bin/console mautic:install --db_port 3306 --db_host db --db_name db --db_user db --db_password db --db_table_prefix ma_ https://mautic.ddev.site/ --env=dev --admin_firstname=admin --admin_lastname=admin --admin_username=admin --admin_email=email@example.com --admin_password=mautic`
4. Notice that the prefix is not honored 

Run this script to reproduce the issue and for quick testing 
```
#!/bin/bash         
mysql -udb -pdb -e "drop database db; create database db"
rm -rf /var/www/html/var/cache/*
rm /var/www/html/app/config/local.php
php bin/console mautic:install --db_port 3306 --db_host db --db_name db --db_user db --db_password db --db_table_prefix ma_ https://mautic.ddev.site/ --env=dev --admin_firstname=admin --admin_lastname=admin --admin_username=admin --admin_email=email@example.com --admin_password=mautic
```

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Try to install with wrong DB settings, no exception should show
3. Add a prefix, database tables should have a prefix

**Please make sure to test this in the following 4 scenarios**:
- CLI installation with a DB prefix
- CLI installation without a DB prefix
- UI installation with a DB prefix
- UI installation without a DB prefix

To clean things up after testing each scenario, you can run `ddev delete --omit-snapshot && rm app/config/local.php && rm -rf var/cache && ddev start` to start with a clean state

### CODE REVIEW

**Comment by dennisameling** [comment]
Tested all 4 scenarios listed above:

With prefix
Install via the UI ✅
Install via CLI ✅

No prefix
Install via the UI ✅
Install via CLI ✅

File: app/bundles/InstallBundle/Command/InstallCommand.php [hunk 1]
```diff
@@ -292,31 +324,31 @@ protected function execute(InputInterface $input, OutputInterface $output)
             }
         }
 
-        if (isset($allParams['site_url']) && !empty($allParams['site_url'])) {
+        if (!empty($allParams['site_url'])) {
```
Comment (dennisameling) [inline app/bundles/InstallBundle/Command/InstallCommand.php:332]: `empty()` first checks if the variable is set, so the `isset()` is superfluous here. That's why I removed it. See https://www.php.net/manual/en/function.empty.php for details.

**Comment by RCheesley** [comment]
Going round in circles trying to work out how to install at CLI without a database prefix. Tried to omit the prefix entry, use "", but not getting anywhere fast. Anyone able to shed some light please?

**Review by RCheesley — CHANGES_REQUESTED** [review]
Seems like the same error is still occurring when installing at the command line without a prefix.

In case it is relevant I am running this on a Mac with Docker Desktop 3.6.0 (67351), DDEV v1.17.7, PHP 7.4/Apache FPM/MariaDB 10.3.

CLI install with prefix: ✅ 
CLI install without prefix: ❌  Schema error (see below)

```
ruth.cheesley@MacBookPro-RuthCheesley mautic % ddev exec bin/console mautic:install --mailer_from_name="DDEV" --mailer_from_email="mautic@ddev.local" --mailer_transport="smtp" --mailer_host="localhost" --mailer_port="1025" --db_driver="pdo_mysql" --db_host="db" --db_port="3306" --db_name="db" --db_user="db" --db_password="db" --admin_email="admin@mautic.local" --admin_password="mautic" https://mautic4.ddev.site --env=dev

 // Clearing the cache for the dev environment with debug true                                                          

                                                                                                                        
 [OK] Cache for the "dev" environment (debug=true) was successfully cleared.                                            
                                                                                                                        

Mautic Install
==============

Parsing options and arguments...
0 - Checking installation requirements...
Ready to Install!
1 - Creating database...
1.1 - Creating schema...
12:15:40 ERROR     [mautic] SCHEMA ERROR: An exception occurred while executing 'SELECT f.alias, f.is_unique_identifer as is_unique, f.type, f.object FROM lead_fields f WHERE f.object = 'lead' ORDER BY f.field_order ASC':

SQLSTATE[42S02]: Base table or view not found: 1146 Table 'db.lead_fields' doesn't exist
1.2 - Loading fixtures...
2 - Creating admin user...
3 - Email configuration...
4 - Final steps...

================
Install complete
================
```
GUI install with prefix ✅ 
GUI install without prefix ✅ 

To add: The table lead_fields does actually exist in the database:

screenshot-mautic4 ddev site_8037-2021 08 22-13_36_42

**Comment by dennisameling** [comment]
@RCheesley I subconsciously ignored `SQLSTATE[42S02]: Base table or view not found: 1146 Table 'db.lead_fields' doesn't exist` as it's not actually an error (see [Alan's comment](https://github.com/mautic/mautic/issues/3436#issuecomment-595752911) on an issue you reported back in 2017). It doesn't show up in the current `features` branch, so let's call it a regression indeed.

I was able to fix the original issue in https://github.com/mautic/mautic/pull/9934, so let's move back to that PR. That way we keep the history intact for the release notes etc.

**Pull request closed without merging**